controllers/argocd/configmap_test.go

```diff
@@ -123,6 +123,101 @@ func TestReconcileArgoCD_reconcileArgoConfigMap(t *testing.T) {
 	}
 }
 
+func TestReconcileArgoCD_reconcileArgoConfigMapWithBanner(t *testing.T) {
+	logf.SetLogger(ZapLogger(true))
+	a := makeTestArgoCD(withBanner("Custom Styles - Banners", ""))
+	r := makeTestReconciler(t, a)
+
+	err := r.reconcileArgoConfigMap(a)
+	assert.NoError(t, err)
+
+	cm := &corev1.ConfigMap{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      common.ArgoCDConfigMapName,
+		Namespace: testNamespace,
+	}, cm)
+	assert.NoError(t, err)
+
+	want := map[string]string{
+		"application.instanceLabelKey": common.ArgoCDDefaultApplicationInstanceLabelKey,
+		"admin.enabled":                "true",
+		"configManagementPlugins":      "",
+		"dex.config":                   "",
+		"ga.anonymizeusers":            "false",
+		"ga.trackingid":                "",
+		"help.chatText":                "Chat now!",
+		"help.chatUrl":                 "https://mycorp.slack.com/argo-cd",
+		"kustomize.buildOptions":       "",
+		"oidc.config":                  "",
+		"repositories":                 "",
+		"repository.credentials":       "",
+		"resource.inclusions":          "",
+		"resource.exclusions":          "",
+		"statusbadge.enabled":          "false",
+		"url":                          "https://argocd-server",
+		"users.anonymous.enabled":      "false",
+		"ui.bannercontent":             "Custom Styles - Banners",
+	}
+
+	if diff := cmp.Diff(want, cm.Data); diff != "" {
+		t.Fatalf("reconcileArgoConfigMap failed:\n%s", diff)
+	}
+}
+
+func TestReconcileArgoCD_reconcileArgoConfigMapWithBannerAndURL(t *testing.T) {
```

Comment:
Just a thought - I think it would be better to have a single test case for this feature, and convert these 2 separate test functions into test cases using table tests ( like https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/util_test.go#L312 for e.g)
This would help avoid code repetition and let us have more targeted and comprehensive unit tests for features

Reply:
@jaideepr97  I simplified it